extensions/amp-animation/0.1/runners/scrolltimeline-worklet-runner.js

```diff
@@ -44,57 +45,49 @@ export class ScrollTimelineWorkletRunner extends AnimationRunner {
     this.players_ = [];
 
     /** @private {number} */
-    this.topRatio_ = viewportData['top-ratio'];
+    this.startScrollOffset_ = viewportData['start-scroll-offset'];
 
     /** @private {number} */
-    this.bottomRatio_ = viewportData['bottom-ratio'];
+    this.endScrollOffset_ = viewportData['end-scroll-offset'];
 
     /** @private {number} */
-    this.topMargin_ = viewportData['top-margin'];
-
-    /** @private {number} */
-    this.bottomMargin_ =
-      viewportData['bottom-margin'];
+    this.initialInViewPercent_ = viewportData['initial-inview-percent'];
   }
 
   /**
   * @override
   * Initializes the players but does not change the state.
    */
   init() {
+    const {documentElement} = this.win_.document;
+    const viewportService = Services.viewportForDoc(documentElement);
+    const scrollSource = viewportService.getScrollingElement();
+
+    const timeRange = getTotalDuration(this.requests_);
+    const adjustedTimeRange = (1 - this.initialInViewPercent_) * timeRange;
+    const initialElementOffset = this.initialInViewPercent_ * timeRange;
```

Comment:
What happens when there's a scroll jump, for example an element get collapsed?

Reply:
Not resolved ( will CC you on a doc that talks about all the issues later this week ). This also is currently a problem with amp-position-observer